sources/close-io/handler.js

```diff
@@ -1,295 +1,58 @@
-    let formattedType = eventBody.event.object_type.split('.')
-    for (let i = 0; i < formattedType.length; i++) {
-      formattedType[i] = formattedType[i].charAt(0).toUpperCase() + formattedType[i].slice(1);
-    }
-    const props = {
-      date_created: eventBody.event.data.date_created,
-      type: eventBody.event.object_type,
-      id: eventBody.event.data.id,
-      created_by: eventBody.event.data.created_by,
-      webhookId: eventBody.subscription_id, // id of the incoming webhook
-      source: 'Close.io'
-    }
-
-    Segment.track({
-      event: formattedType.join(' ') + ' Created',
-      userId: eventBody.event.data.created_by,
-      properties: props
-    })
-  }
-
-  if (eventBody.event.action == 'updated') {
-    let formattedType = eventBody.event.object_type.split('.')
-    for (let i = 0; i < formattedType.length; i++) {
-      formattedType[i] = formattedType[i].charAt(0).toUpperCase() + formattedType[i].slice(1);
-    }
-    const props = {
-      date_updated: eventBody.event.data.date_updated,
-      date_created: eventBody.event.data.date_created,
-      type: eventBody.event.object_type,
-      id: eventBody.event.data.id,
-      created_by: eventBody.event.data.created_by,
-      webhookId: eventBody.subscription_id, // id of the incoming webhook
-      source: 'Close.io'
-    }
-
-    Segment.track({
-      event: formattedType.join(' ') + ' Updated',
-      userId: eventBody.event.data.created_by,
-      properties: props
-    })
-  }
-
-  if (eventBody.event.action == 'deleted') {
-    let formattedType = eventBody.event.object_type.split('.')
-    for (let i = 0; i < formattedType.length; i++) {
-      formattedType[i] = formattedType[i].charAt(0).toUpperCase() + formattedType[i].slice(1);
-    }
-    const props = {
-      date_updated: eventBody.event.data.date_updated,
-      date_created: eventBody.event.data.date_created,
-      type: eventBody.event.object_type,
-      id: eventBody.event.data.id,
-      created_by: eventBody.event.data.created_by,
-      webhookId: eventBody.subscription_id, // id of the incoming webhook
-      source: 'Close.io'
-    }
-
-    Segment.track({
-      event: formattedType.join(' ') + ' Deleted',
-      userId: eventBody.event.data.created_by,
-      properties: props
-    })
-  }
+  const { event: body } = request.json();
+  const {
+    object_type,
+    subscription_id,
+    data: { id, created_by, updated_by },
+    data,
+    action
+  } = body;
+  const sourceCtx = () => ({ integration: { name: "Close.io [Custom]" } });
+
+  const objectTypeCollectionMapping = new Map([
+    ["opportunity", "opportunities"],
+    ["lead", "leads"],
+    ["contact", "contacts"],
+    ["activity.note", "activity_notes"],
+    ["activity.call", "activity_calls"],
+    ["activity.email", "activity_emails"]
+  ]);
+
+  const collection = objectTypeCollectionMapping.get(object_type);
+  if (!collection) return;
+
+  let properties = {
+    ...data,
+    ...subscription_id
```

Comment:
similarly here I don't think this `...subscription_id` line will work as-is and could probably just be removed
```suggestion
```